ObjCTraining/AddPropertyInfoViewController.m

```diff
@@ -0,0 +1,331 @@
+}
+
+- (IBAction)buttonContinueTapped:(UIButton *)sender {
+    if ([self checkInputData]){
+        NSLog(@"GO NEXT");
+    }
+}
+
+#pragma mark - UITextFieldDelegate
+- (BOOL) textFieldShouldReturn:(UITextField *)textField
+{
+    NSInteger nextTag = textField.tag + 1;
+    UIResponder* nextResponder = [textField.superview viewWithTag:nextTag];
+    if (nextResponder) {
+        [nextResponder becomeFirstResponder];
+    } else {
+        [textField resignFirstResponder];
+    }
+    return NO;
+}
+
+#pragma mark - Validation
+- (BOOL) isAnyFieldEmpty {
+    BOOL isAnyfieldEmpty = NO;
+    for (id key in isFieldsEmpty){
+        //Address2Field could be empty
+        if([[isFieldsEmpty objectForKey:key] boolValue] && [key integerValue] != Address2FieldTag){
+            [self invalidateFieldForTag:[key integerValue] isValid:NO];
+            isAnyfieldEmpty = YES;
+        }
+    }
+    return isAnyfieldEmpty;
+}
+
+- (BOOL) isAnyFieldWrong {
+    for (id key in isFieldsWrong){
+        if([[isFieldsWrong objectForKey:key] boolValue]){
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (void) invalidateFieldForTag: (NSInteger) tag isValid: (BOOL) isValid {
+    switch (tag) {
+        case Address1FieldTag:
+            [self updateField: self.address1TextField textLabel: self.address1Title isCorrect: isValid];
+            break;
+        case Address2FieldTag:
+            [self updateField: self.address2TextField textLabel: self.address2Title isCorrect: isValid];
+            break;
+        case CityFieldTag:
+            [self updateField: self.cityTextField textLabel: self.cityTitle isCorrect: isValid];
+            break;
+        case StateFieldTag:
+            [self updateField: self.stateTextField textLabel: self.stateTitle isCorrect: isValid];
+            break;
+        case ZipFieldTag:
+            [self updateField: self.zipTextField textLabel: self.zipTitle isCorrect: isValid];
+            break;
+        case CountryFieldTag:
+            [self updateField: self.countryTextField textLabel: self.countryTitle isCorrect: isValid];
+            break;
+    }
+}
+
+- (BOOL) checkInputData {
+    BOOL isFieldEmpty = [self isAnyFieldEmpty];
+    BOOL isFieldWrong = [self isAnyFieldWrong];
+    if(isFieldEmpty || isFieldWrong) {
+        [self showErroMessage: [self creatErrorMessage: isFieldEmpty isAnyFieldWrong: isFieldWrong]];
+        return NO;
+    } else {
+        [self clearErroMessage];
+        return YES;
+    }
+}
+
+- (void) clearErroMessage {
+    self.errorMessage.hidden = true;
+    self.ErrorMessageHeightConstraint.constant = 0;
+}
+
+- (void) showErroMessage: (NSString *) errorMessage {
+    self.errorMessage.hidden = false;
+    self.errorMessage.text = errorMessage;
+    self.ErrorMessageHeightConstraint.constant = 21;
+}
+
+- (void) updateField: (UITextField *)textField textLabel: (UILabel *)textLabel isCorrect: (BOOL) isCorrect {
```

Comment:
Again, need to rename. Methods should be like sentences. 
- (void)updateTextField:(UITextField *) withLabel(UILabel *)label style :(*add style enum)style